src/Fixer/Whitespace/UnionTypesSpacesFixer.php

```diff
@@ -0,0 +1,119 @@
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class UnionTypesSpacesFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'A single space or none should be around union type operator.',
+            [
+                new CodeSample(
+                    "<?php\ntry\n{\n    new Foo();\n} catch (ErrorA | ErrorB \$e) {\necho'error';}\n"
+                ),
+                new CodeSample(
+                    "<?php\ntry\n{\n    new Foo();\n} catch (ErrorA|ErrorB \$e) {\necho'error';}\n",
+                    ['space' => 'single']
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction foo(int | string \$x)\n{\n}\n",
+                    new VersionSpecification(80000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(CT::T_TYPE_ALTERNATION);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run after BinaryOperatorSpacesFixer.
+     */
+    public function getPriority(): int
+    {
+        return -33;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('space', 'spacing to apply around union type operator.'))
+                ->setAllowedValues(['none', 'single'])
+                ->setDefault('none')
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(CT::T_TYPE_ALTERNATION)) {
+                continue;
+            }
+
+            if ('single' === $this->configuration['space']) {
+                $this->ensureSingleSpace($tokens, $index + 1, 0);
```

Comment:
I'm pretty sure you find those test cases boring just as I do but... this will fail in case the previous/next index is a comment instead of a whitespace. Please add test like
```php
Foo/* not a space */|/* not a space */Bar

Foo// not a space
|//not a space
Bar
```